contracts/interfaces/registries/ILicenseRegistryV2.sol

```diff
@@ -0,0 +1,168 @@
+
+    /// @notice Registers a new license template in the Story Protocol.
+    /// @param licenseTemplate The address of the license template to register.
+    function registerLicenseTemplate(address licenseTemplate) external;
+
+    /// @notice Checks if a license template is registered.
+    /// @param licenseTemplate The address of the license template to check.
+    /// @return Whether the license template is registered.
+    function isRegisteredLicenseTemplate(address licenseTemplate) external view returns (bool);
+
+    /// @notice Registers a derivative IP and its relationship to parent IPs.
+    /// @param ipId The address of the derivative IP.
+    /// @param parentIpIds An array of addresses of the parent IPs.
+    /// @param licenseTemplate The address of the license template used.
+    /// @param licenseTermsIds An array of IDs of the license terms.
+    function registerDerivativeIp(
+        address ipId,
+        address[] calldata parentIpIds,
+        address licenseTemplate,
+        uint256[] calldata licenseTermsIds
+    ) external;
+
+    /// @notice Checks if an IP is a derivative IP.
+    /// @param ipId The address of the IP to check.
+    /// @return Whether the IP is a derivative IP.
+    function isDerivativeIp(address ipId) external view returns (bool);
+
+    /// @notice Checks if an IP has derivative IPs.
+    /// @param ipId The address of the IP to check.
+    /// @return Whether the IP has derivative IPs.
+    function hasDerivativeIps(address ipId) external view returns (bool);
+
+    /// @notice Verifies the minting of a license token.
+    /// @param licensorIpId The address of the licensor IP.
+    /// @param licenseTemplate The address of the license template where the license terms are defined.
+    /// @param licenseTermsId The ID of the license terms will mint the license token.
+    /// @param isMintedByIpOwner Whether the license token is minted by the IP owner.
+    /// @return The configuration for minting the license.
+    function verifyMintLicenseToken(
+        address licensorIpId,
+        address licenseTemplate,
+        uint256 licenseTermsId,
+        bool isMintedByIpOwner
+    ) external view returns (Licensing.MintingLicenseConfig memory);
+
+    /// @notice Attaches license terms to an IP.
+    /// @param ipId The address of the IP to which the license terms are attached.
+    /// @param licenseTemplate The address of the license template.
+    /// @param licenseTermsId The ID of the license terms.
+    function attachLicenseTermsToIp(address ipId, address licenseTemplate, uint256 licenseTermsId) external;
+
+    /// @notice Checks if license terms exist.
+    /// @param licenseTemplate The address of the license template where the license terms are defined.
+    /// @param licenseTermsId The ID of the license terms.
+    /// @return Whether the license terms exist.
+    function exists(address licenseTemplate, uint256 licenseTermsId) external view returns (bool);
```

Comment:
better use a more descriptive name?

exists => IsLicenseTermDefined

Reply:
Feel "exists()" has descriptive enough.  it is common practice that use "exists" to check an object/token does exists or not. developer easy to understand. 

Lens also use `exists()` to check `profile` exists or not. 

```solidity
    function exists(uint256 profileId) internal view returns (bool) {
        return StorageLib.getTokenData(profileId).owner != address(0);
    }
```